script.js

```diff
@@ -0,0 +1,248 @@
+console.log("hi");
+// var count=0;
+class Game {
+    constructor(canvas) {
+        this.canvas = canvas;
+        this.screen = canvas.getContext("2d");
+        this.player = new Player(this);
+        this.coin = new Coin(this);
+        this.badCoin = new BadCoin(this);
+        this.badCoinB = new BadCoinB(this);
```

Comment:
We talked about this in class -- if you have two classes that are very close to each other in implementation, either pass in the info that differs between them, or consider subclasses.